contracts/exchange-wrappers/ZeroExExchangeWrapper.sol

```diff
@@ -13,38 +13,104 @@
 
 pragma solidity ^0.8.0;
 
+import { I_ExchangeWrapper } from "../interfaces/I_ExchangeWrapper.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { I_StarkwareContract } from "../interfaces/I_StarkwareContracts.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @title ZeroExExchangeWrapper
+ * @author dYdX
+ *
+ * Wrapper around 0x API ERC20 token swap.
+ */
+contract ZeroExExchangeWrapper is I_ExchangeWrapper {
+    using SafeERC20 for IERC20;
+
+    // ============ State Variables ============
+
+    IERC20 immutable USDC_ADDRESS;
+
+    uint256 immutable USDC_ASSET_TYPE;
+
+    I_StarkwareContract public immutable STARKWARE_CONTRACT;
+
+    // ============ Constructor ============
+
+    constructor(
+        I_StarkwareContract starkwareContractAddress,
+        IERC20 usdcAddress,
+        uint256 usdcAssetType
+    )
+    {
+        STARKWARE_CONTRACT = starkwareContractAddress;
+        USDC_ADDRESS = usdcAddress;
+        USDC_ASSET_TYPE = usdcAssetType;
+
+        // Set the allowance to the highest possible value.
+        usdcAddress.safeApprove(address(starkwareContractAddress), type(uint256).max);
```

Comment:
Sorry if this is back-and-forth, but this contract should just be an exchange wrapper and shouldn't be aware of Starkware. It shouldn't do the deposit. The exchange wrapper should be reusable for both deposit, withdraw, or any other operations.

For the same reason, this contract doesn't need to be aware of USDC either.

The approve logic can go in the CurrencyConverter.